### PR TITLE
[Parser] Check for postfix operator in import paths

### PR DESCRIPTION
**Pull request by angela-laar**
Fixes rdar://87159294

### CODE REVIEW

**Comment by angela-laar** [comment]
@swift-ci Please smoke test

File: include/swift/AST/DiagnosticsParse.def [hunk 1]
```diff
@@ -1616,6 +1616,8 @@ ERROR(attr_private_import_expected_sourcefile_name,none,
       "expected a source file name in @_private(sourceFile:)", ())
 ERROR(attr_private_import_expected_colon,none,
       "expected ':' after @_private(sourceFile", ())
+ERROR(operator_in_import_path,none,
+      "Cannot include postfix operator in import declaration.",())
```
Comment (xwu) [inline include/swift/AST/DiagnosticsParse.def:1620]: Error messages don't have capitalization or trailing punctuation:

```suggestion
      "cannot include postfix operator in import declaration",())
```

File: include/swift/AST/DiagnosticsParse.def [hunk 2]
```diff
@@ -1616,6 +1616,8 @@ ERROR(attr_private_import_expected_sourcefile_name,none,
       "expected a source file name in @_private(sourceFile:)", ())
 ERROR(attr_private_import_expected_colon,none,
       "expected ':' after @_private(sourceFile", ())
+ERROR(operator_in_import_path,none,
```
Comment (xwu) [inline include/swift/AST/DiagnosticsParse.def:1619]: Maybe something like `unexpected_operator_in_import_decl`? Would not group this here with diagnostics for the underscored `@_private` import feature but with other diagnostics for regular import declarations.

Comment (angela-laar) [reply include/swift/AST/DiagnosticsParse.def]: I like that, I settled on `unexpected_operator_in_import_path` and I've moved the diagnostic to more similar ones in my newest commit

**Comment by angela-laar** [comment]
@swift-ci Please smoke test

**Comment by angela-laar** [comment]
@swift-ci Please smoke test

**Comment by angela-laar** [comment]
@swift-ci  Please clean smoke test macOS platform

**Review by kavon — APPROVED** [review]
👍🏼 this looks great! thank you.

**Comment by angela-laar** [comment]
@swift-ci Please smoke test macOS platform

**Comment by angela-laar** [comment]
@swift-ci Please smoke test macOS platform

**Comment by angela-laar** [comment]
@swift-ci Please smoke test macOS platform

**Comment by angela-laar** [comment]
@swift-ci Please smoke test

**Pull request merged**